### ISSUE TITLE
Versioning: show version on Form Builder summary page

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#1795]
Currently, the Form Builder Summary page doesn't show any published form version information.

One reason for this is that Form Builder deals with data stored in `orbeon_form_data`, and only pushes to `orbeon_form_definition` upon publish. So it doesn't have any information about published forms. This has been found confusing to some, BTW.

One way to change this would be to have FB only deal with `orbeon_form_definition`. We would need to:
- support the notion of an "in progress" form definition
- also support regular FR autosave drafts
- support Summary page access in particular search

However, this is a big task.

In the meanwhile, we could choose a simpler path, as  follows:
- form definitions keep a new "version" metadata
- this would show on the FB Summary page in a new column
- initially, this would be N/A
- upon Publish
  - about to be published version is stored in metadata
  - form definition is saved (2016-03-21: Since 2016.1 we save the form definition just before publishing.)
  - form is published
    - Q: What if this fails? Shouldn't have version metadata saved.
- upon editing
  - if there is no existing version, nothing to do
  - if there is an existing version, as soon as the user makes a change to the form, add a "changed" flag to indicate that the form has been changed
- Summary page shows column
  - N/A
  - 1
  - 1 with changes
  - 2
  - 2 with changes
  - etc.
  - only one row per app/form name at this time (although user can still duplicate / create new rows with same app/form name, see #818)
- edit source
  - we don't want to mess with the version by mistake here
  - before replacing source, copy form version metadata aside, and restore it
  - possibly allow user to edit last published version in edit form metadata dialog?

**Comment by ebruchez** [comment]
[+1 from customer](https://basecamp.com/1721271/projects/5657877/messages/36056592)

**Comment by ebruchez** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/11457615/messages/1914475131)

**Comment by ebruchez** [comment]
We still have two possibilities, possibly not exclusive:

1. What this issue is about, namely store a `last-published-version` metadata in the form definition. This has the benefit of simplicity, but the drawback of possible in correctness, for example if a form definition is published separately via the Publish API instead of directly from Form Builder.
2. Using the Form Metadata API in the Form Builder Summary page to show the actual information about published form definitions and merge it.

Solution 1 appeared easier to implement, which is why we still considered it. See also [this comment](/orbeon/orbeon-forms/issues/1138#issuecomment-680995505) on #1138 regarding using stages.

**Comment by avernet** [comment]
About option 1 in https://github.com/orbeon/orbeon-forms/issues/1795#issuecomment-1086395852, I also favor ease of implementation, but it looks to me like such denormalization would create a number of issues down the road, in what we could think are corner cases, and I'd bet sure some customers will hit those corners ;). For instance, this information stored in the "Form Builder form definition" might get outdated on a remote when a new form definition is published from the Form Runner home page, or something changes at the databases-level or database-configuration-level (`properties-local.xml` changed to point to a different database for published forms, backup of published forms restored…).

tl;dr -1 on storing `last-published-version` metadata in the form definition

**Comment by ebruchez** [comment]
+1 today about this issue in connection with library forms, see #6684.
